src/lib/Ouroboros/Byron/Proxy/Index/ChainDB.hs

```diff
@@ -85,7 +87,8 @@ trackChainDB rr idx cdb = bracket acquireReader releaseReader $ \rdr -> do
   -- ... then attempt to stay in sync.
   trackReaderBlocking idx rdr
   where
-  acquireReader :: IO (Reader IO blk (Header blk))
-  acquireReader = ChainDB.deserialiseReader <$> ChainDB.newHeaderReader cdb rr
-  releaseReader :: Reader IO blk (Header blk) -> IO ()
+  acquireReader :: IO (Reader IO blk (IO (Header blk)))
+  --acquireReader = ChainDB.deserialiseReader <$> ChainDB.newReader cdb rr component
+  acquireReader = ChainDB.newReader cdb rr GetHeader
```

Comment:
To avoid propagating the `m`:
```suggestion
  acquireReader = ChainDB.traverseReader id <$> ChainDB.newReader cdb rr GetHeader
```

Reply:
Oh nice. I was a bit annoyed by that but didn't know about `traverseReader`